File: app/src/main/java/io/github/marcocipriani01/telescopetouch/activities/CompassCalibrationModule.java
package io.github.marcocipriani01.telescopetouch.activities;

import android.content.Context;

import dagger.Module;
import dagger.Provides;
import io.github.marcocipriani01.telescopetouch.inject.PerActivity;

/**
 * Created by johntaylor on 4/24/16.
 */
@Module
public class CompassCalibrationModule {
    private final CompassCalibrationActivity activity;

    public CompassCalibrationModule(CompassCalibrationActivity activity) {
        this.activity = activity;
    }

    @Provides
    @PerActivity
    Context provideContext() {
        return activity;
    }
}

File: app/src/main/java/com/google/android/stardroid/activities/CompassCalibrationModule.kt
package com.google.android.stardroid.activities

import android.content.Context
import com.google.android.stardroid.activities.CompassCalibrationActivity
import com.google.android.stardroid.inject.PerActivity
import dagger.Module
import dagger.Provides

/**
 * Created by johntaylor on 4/24/16.
 */
@Module
class CompassCalibrationModule(private val activity: CompassCalibrationActivity) {
    @Provides
    @PerActivity
    fun provideContext(): Context {
        return activity
    }
}